### PR TITLE
Windows: Split ContainerExecCreate

### PR DESCRIPTION
**Pull request by lowenna**
Signed-off-by: John Howard jhoward@microsoft.com

Closes #12760

### CODE REVIEW

**Comment by jhowardmsft** [comment]
Redoing 12760 where github wasn't showing up the updated changes for some reason.

File: daemon/exec_windows.go [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// +build windows
+
+package daemon
+
+// checkExecSupport returns an error if the exec driver does not support exec
+func checkExecSupport(DriverName string) error {
+	return nil
```
Comment (tiborvass) [inline daemon/exec_windows.go]: how about sending a `errors.New("exec unsupported")` ?

Comment (jhowardmsft) [inline daemon/exec_windows.go]: Exec is supported by the Windows driver, hence returning nil is the right thing here.

File: daemon/exec.go [hunk 2]
```diff
@@ -111,8 +110,9 @@ func (d *Daemon) getActiveContainer(name string) (*Container, error) {
 
 func (d *Daemon) ContainerExecCreate(config *runconfig.ExecConfig) (string, error) {
 
-	if strings.HasPrefix(d.execDriver.Name(), lxc.DriverName) {
-		return "", lxc.ErrExec
+	// Not all drivers support Exec (LXC for example)
+	if execSupportCheck := checkExecSupport(d.execDriver.Name()); execSupportCheck != nil {
```
Comment (tiborvass) [inline daemon/exec.go]: Should be `err` or should contain it.

Comment (jhowardmsft) [inline daemon/exec.go]: Fixed :)

File: daemon/exec_windows.go (hunk 1 above)
Comment (ahmetalpbalkan) [inline daemon/exec_windows.go]: @jhowardmsft we could add a note to the godoc and finish the sentence like:

    ... or nil if it is supported.

Comment (jhowardmsft) [inline daemon/exec_windows.go]: Updated

**Comment by vieux** [comment]
LGTM

**Comment by crosbymichael** [comment]
LGTM

**Pull request merged**